pennylane/gradients/gradient_transform.py

```diff
@@ -20,6 +20,60 @@
 from pennylane.transforms.tape_expand import expand_invalid_trainable
 
 
+def gradient_analysis(tape, use_graph=True, grad_fn=None):

```

Comment:
This function was moved from `pennylane.gradients.parameter_shift._gradient_analysis`, since it turns out that `finite_diff` also needed to use it.

Reply:
Out of scope: the name `gradient_analysis` seems to suggest that we're getting information about the gradient of the tape.

At the same time:
> Note that this function modifies the input tape in-place.

I think seeing a line like
```python
gradient_analysis(tape, grad_fn=finite_diff)
```
and mutating the tape data under the hood is unexpected. Any reasons why the `_update_gradient_info` name was abandoned? It contained a verb that referred to the updating.

Reply:
@antalszava yep 😆 Agree on all points. `gradient_analysis` is actually pretty much a 1-1 port of `tape._update_gradient_info()`, hence why it is in-place - it was originally a Tape method!

In terms of why it was renamed, I can't remember. Happy for it's name to be changed, but would rather it be done in a separate PR

Reply:
Although rather than renaming it, we should probably refactor this function to (a) not be in place, and (b) have a smarter/cleaner form of caching rather than adding a new attribute to the tape.